[clang-tidy][performance-move-const-arg] Fix crash when argument type has no definition

nicovank opened the pull request:
Fix #111450.

When the argument type is forward-declared and there is no definition, `hasMoveConstructor` triggers the assert here:
https://github.com/llvm/llvm-project/blob/7f65377880ce6a0e5eaa4cb2591b86b8c8a24ee6/clang/include/clang/AST/DeclCXX.h#L465
https://github.com/llvm/llvm-project/blob/7f65377880ce6a0e5eaa4cb2591b86b8c8a24ee6/clang/include/clang/AST/DeclCXX.h#L864

Check `hasDefinition()` before `hasMoveConstructor()`.

5chmidti reviewed (approved):
Please add a release note, otherwise LGTM

nicovank committed:
[clang-tidy][performance-move-const-arg] Fix crash when argument type has no definition

Fix #111450.

When the argument type is forward-declared and there is no definition, `hasMoveConstructor` triggers the assert here:
https://github.com/llvm/llvm-project/blob/7f65377880ce6a0e5eaa4cb2591b86b8c8a24ee6/clang/include/clang/AST/DeclCXX.h#L465
https://github.com/llvm/llvm-project/blob/7f65377880ce6a0e5eaa4cb2591b86b8c8a24ee6/clang/include/clang/AST/DeclCXX.h#L864

Check `hasDefinition()` before `hasMoveConstructor()`.
